packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -113,116 +113,102 @@ export const satisfies = (zcf, seat, update) => {
     );
   }
 
   // Check whether reallocate would error before calling. If
   // it would error, reject the right offer and return.
-  const offerSafeForLeft = keepLeft.seat.isOfferSafe(leftAllocation);
-  const offerSafeForRight = tryRight.seat.isOfferSafe(rightAllocation);
+  const offerSafeForLeft = left.seat.isOfferSafe(leftAllocation);
+  const offerSafeForRight = right.seat.isOfferSafe(rightAllocation);
   if (!(offerSafeForLeft && offerSafeForRight)) {
-    console.log(`currentLeftAllocation`, keepLeft.seat.getCurrentAllocation());
-    console.log(`currentRightAllocation`, tryRight.seat.getCurrentAllocation());
+    console.log(`currentLeftAllocation`, left.seat.getCurrentAllocation());
+    console.log(`currentRightAllocation`, right.seat.getCurrentAllocation());
     console.log(`proposed left reallocation`, leftAllocation);
     console.log(`proposed right reallocation`, rightAllocation);
     // show the constraints
-    console.log(`left want`, keepLeft.seat.getProposal().want);
-    console.log(`right want`, tryRight.seat.getProposal().want);
+    console.log(`left want`, left.seat.getProposal().want);
+    console.log(`right want`, right.seat.getProposal().want);
 
     if (!offerSafeForLeft) {
       console.log(`offer not safe for left`);
     }
     if (!offerSafeForRight) {
       console.log(`offer not safe for right`);
     }
-    throw tryRight.seat.kickOut(
-      new Error(
-        `The trade between left ${keepLeft} and right ${tryRight} failed offer safety. Please check the log for more information`,
-      ),
+    throw new Error(
+      `The trade between left ${left} and right ${right} failed offer safety. Please check the log for more information`,
     );
   }
 
   return zcf.reallocate(
-    keepLeft.seat.stage(leftAllocation),
-    tryRight.seat.stage(rightAllocation),
+    left.seat.stage(leftAllocation),
+    right.seat.stage(rightAllocation),
   );
 };
 
-/**
- * If the two handles can trade, then swap their compatible assets,
- * marking both offers as complete.
- *
- * The surplus remains with the original offer. For example if
- * offer A gives 5 moola and offer B only wants 3 moola, offer A
- * retains 2 moola.
- *
- * If the keep offer is no longer active (it was already completed), the try
- * offer will be rejected with a message (provided by 'keepHandleInactiveMsg').
- *
- * TODO: If the try offer is no longer active, swap() should terminate with
- * a useful error message.
- *
- * If the swap fails, no assets are transferred, and the 'try' offer is rejected.
- *
- * @param {ContractFacet} zcf
- * @param {ZCFSeat} keepSeat
- * @param {ZCFSeat} trySeat
- * @param {String} [keepHandleInactiveMsg]
- */
+/** @type Swap */
 export const swap = (
   zcf,
-  keepSeat,
-  trySeat,
-  keepHandleInactiveMsg = 'prior offer is unavailable',
+  leftSeat,
+  rightSeat,
+  leftHasExitedMsg = 'the left seat in swap() has exited',
```

Comment:
Given async, let's add several elements to the error about e.g., what brands are being traded?

Reply:
Hmm, we can't currently display brands. What kind of information would you like to see?

Reply:
As an aside, all of the information console logged within `trade` applies to `swap` too because swap uses trade.